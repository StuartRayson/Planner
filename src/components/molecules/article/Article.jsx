import classNames from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';

import RichText from '../../atoms/richText/RichText.jsx';
import Heading from '../../atoms/heading/Heading.jsx';

import './article.scss';

const Article = ({ className, title, content }) => {
  const classes = classNames(
    'm-article',
    className
  );

  return (
    <article className={classes}>
      <Heading
        type="h2"
        text={title}
        className="m-article__title"
      />
      <div className="m-article__body">
        <RichText html={content} />
      </div>
    </article>
  );
};

Article.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Article;
