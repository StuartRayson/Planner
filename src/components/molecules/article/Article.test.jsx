import React from 'react';
import { shallow } from 'enzyme';

import Article from './Article.jsx';

const config = {
  title: 'title',
  content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'
};

describe('<Article />', () => {
  it('renders without crashing', () => {
    shallow(<Article {...config} />);
  });

  it('should render and match snapshot with basic config', () => {
    const wrapper = shallow(
      <Article {...config} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
