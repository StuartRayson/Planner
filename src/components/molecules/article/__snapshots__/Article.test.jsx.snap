// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> should render and match snapshot with basic config 1`] = `
<article
  className="m-article"
>
  <Heading
    className="m-article__title"
    text="title"
    type="h2"
  />
  <div
    className="m-article__body"
  >
    <RichText
      html="Lorem ipsum dolor sit amet, consectetur adipiscing elit."
    />
  </div>
</article>
`;
